pip_audit/_dependency_source/requirement.py

```diff
@@ -84,10 +84,40 @@ def collect(self) -> Iterator[Dependency]:
         Raises a `RequirementSourceError` on any errors.
         """
 
+        with ExitStack() as stack:
+            # We need to handle process substitution inputs so we can invoke
+            # `pip-audit` like so:
+            #
+            #   pip-audit -r <(echo 'something')
+            #
+            # Since `/dev/fd/<n>` inputs are unique to the parent process, we
+            # can't pass these file names to `pip` and expect `pip` to able to
+            # read them.
+            #
+            # In order to get around this, we're going to copy each input into a
+            # a corresponding temporary file and then pass that set of files
+            # into `pip`.
+            tmp_files: list[IO[str]] = [
+                stack.enter_context(NamedTemporaryFile(mode="w")) for _ in self._filenames
+            ]
+
+            # For each input file, copy it to one of our temporary files.
+            # Ensure we flush and fsync so our writes are visible to `pip`.
+            for filename, tmp_file in zip(self._filenames, tmp_files):
+                with filename.open("r") as f:
+                    shutil.copyfileobj(f, tmp_file)
+                    tmp_file.flush()
+                    os.fsync(tmp_file)
```

Comment:
Do we need both of these? I would expect one or the other to be sufficient, but POSIX I/O never fails to surprise...

Reply:
Oh good point. I think those are both required for the write to be durable, but we don't care about the temporary file persisting if we power off anyway. I think I can get away with just `flush`.

Reply:
[3bb322b](https://github.com/pypa/pip-audit/pull/538/commits/3bb322bc905f6fcc8f53ca3787abfd7a25011b45)